access/resource_ip_access_list.go

```diff
@@ -30,48 +33,86 @@ func ResourceIPAccessList() common.Resource {
 	})
 	return common.Resource{
 		Schema: s,
+		CanSkipReadAfterCreateAndUpdate: func(d *schema.ResourceData) bool {
+			//only skip read after create
+			return d.IsNewResource()
+		},
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			w, err := c.WorkspaceClient()
-			if err != nil {
-				return err
-			}
 			var iacl settings.CreateIpAccessList
+			var updateIacl settings.UpdateIpAccessList
 			common.DataToStructPointer(d, s, &iacl)
-			status, err := w.IpAccessLists.Create(ctx, iacl)
-			if err != nil {
-				return err
-			}
-			d.SetId(status.IpAccessList.ListId)
-			return nil
+			common.DataToStructPointer(d, s, &updateIacl)
+			return c.AccountOrWorkspaceRequest(func(acc *databricks.AccountClient) error {
+				status, err := acc.IpAccessLists.Create(ctx, iacl)
+				if err != nil {
+					return err
+				}
+				ipAclId := status.IpAccessList.ListId
+				// need to wait until the ip access list is available from get
+				retry.RetryContext(ctx, 10*time.Minute, func() *retry.RetryError {
+					_, err := acc.IpAccessLists.GetByIpAccessListId(ctx, ipAclId)
+					var apiErr *apierr.APIError
+					if !errors.As(err, &apiErr) {
+						return retry.NonRetryableError(err)
+					}
+					if apiErr.StatusCode == 404 {
+						return retry.RetryableError(err)
+					} else {
+						return retry.NonRetryableError(err)
+					}
+				})
+				//need to enable the IP Access List with update
+				if d.Get("enabled").(bool) {
+					updateIacl.IpAccessListId = ipAclId
+					err = acc.IpAccessLists.Update(ctx, updateIacl)
+					if err != nil {
+						return err
+					}
+				}
+				d.SetId(ipAclId)
```

Comment:
I think that we need to add `StructToData` on the result of the `acc.IpAccessLists.Create` here (and same for workspace-level create)

Reply:
so the issue, we can only enable the Account-level IP ACL after it has been created, so we still need to wait until it has been created. Otherwise the call to enable the ACL will fail